File: server/Netptune.Core/Enums/ProjectTaskStatus.cs
namespace Netptune.Core.Enums;

public enum ProjectTaskStatus
{
    New = 0,

    Complete = 1,

    InProgress = 2,

    OnHold = 3,

    UnAssigned = 4,

    AwaitingClassification = 5,

    InActive = 6,
}

public static class ProjectTaskStatusExtensions
{
    public static BoardGroupType GetGroupTypeFromTaskStatus(this ProjectTaskStatus status)
    {
        return status switch
        {
            ProjectTaskStatus.New => BoardGroupType.Todo,
            ProjectTaskStatus.InActive => BoardGroupType.Todo,
            ProjectTaskStatus.Complete => BoardGroupType.Done,
            ProjectTaskStatus.AwaitingClassification => BoardGroupType.Todo,
            ProjectTaskStatus.UnAssigned => BoardGroupType.Backlog,
            _ => BoardGroupType.Backlog,
        };
    }
}

File: client/src/app/core/enums/project-task-status.ts
export enum TaskStatus {
  new = 0,
  complete = 1,
  inProgress = 2,
  onHold = 3,
  unAssigned = 4,
  awaitingClassification = 5,
  inActive = 6,
}

export const taskStatusLabels: Record<TaskStatus, string> = {
  [TaskStatus.new]: 'New',
  [TaskStatus.complete]: 'Complete',
  [TaskStatus.inProgress]: 'In Progress',
  [TaskStatus.onHold]: 'On Hold',
  [TaskStatus.unAssigned]: 'Un-assigned',
  [TaskStatus.awaitingClassification]: 'Awaiting Classification',
  [TaskStatus.inActive]: 'Inactive',
};

export const taskStatusOptions = [
  TaskStatus.new,
  TaskStatus.inProgress,
  TaskStatus.onHold,
  TaskStatus.unAssigned,
  TaskStatus.awaitingClassification,
  TaskStatus.inActive,
  TaskStatus.complete,
];
